services/venice-controller/src/main/java/com/linkedin/venice/controller/server/CreateVersion.java

```diff
@@ -71,6 +74,356 @@ public CreateVersion(
+          response.getKafkaBootstrapServers(),
+          request.getPushJobId(),
+          request.getStoreName(),
+          request.getClusterName());
+    }
+  }
+
+  static CompressionStrategy getCompressionStrategy(Version version, String responseTopic) {
+    if (Version.isRealTimeTopic(responseTopic)) {
+      return CompressionStrategy.NO_OP;
+    }
+    return version.getCompressionStrategy();
+  }
+
+  static String determineResponseTopic(String storeName, Version version, PushType pushType) {
+    String responseTopic;
+    if (pushType == PushType.INCREMENTAL) {
+      // If incremental push with a dedicated real-time topic is enabled then use the separate real-time topic
+      if (version.isSeparateRealTimeTopicEnabled()) {
+        responseTopic = Version.composeSeparateRealTimeTopic(storeName);
+      } else {
+        responseTopic = Version.composeRealTimeTopic(storeName);
+      }
+    } else if (pushType == PushType.STREAM) {
+      responseTopic = Version.composeRealTimeTopic(storeName);
+    } else if (pushType == PushType.STREAM_REPROCESSING) {
+      responseTopic = Version.composeStreamReprocessingTopic(storeName, version.getNumber());
+    } else {
+      responseTopic = version.kafkaTopicName();
+    }
+    return responseTopic;
+  }
+
+  private void handleNonStreamPushType(
+      Admin admin,
+      Store store,
+      RequestTopicForPushRequest request,
+      VersionCreationResponse response,
+      Lazy<Boolean> isActiveActiveReplicationEnabledInAllRegions) {
+    String clusterName = request.getClusterName();
+    String storeName = request.getStoreName();
+    PushType pushType = request.getPushType();
+    // Check if requestTopicForPush can be handled by child controllers for the given store
+    if (!admin.whetherEnableBatchPushFromAdmin(storeName)) {
+      throw new VeniceUnsupportedOperationException(
+          request.getPushType().name(),
+          "Please push data to Venice Parent Colo instead");
+    }
+    final Version version = admin.incrementVersionIdempotent(
+        clusterName,
+        storeName,
+        request.getPushJobId(),
+        response.getPartitions(),
+        response.getReplicas(),
+        pushType,
+        request.isSendStartOfPush(),
+        request.isSorted(),
+        request.getCompressionDictionary(),
+        Optional.ofNullable(request.getSourceGridFabric()),
+        Optional.ofNullable(request.getCertificateInRequest()),
+        request.getRewindTimeInSecondsOverride(),
+        Optional.ofNullable(request.getEmergencySourceRegion()),
+        request.isDeferVersionSwap(),
+        request.getTargetedRegions(),
+        request.getRepushSourceVersion());
+
+    // If Version partition count different from calculated partition count use the version count as store count
+    // may have been updated later.
+    if (version.getPartitionCount() != response.getPartitions()) {
```

Comment:
should here we log the partition count change so that we won't have to scratch our head in future troubleshoot?

Reply:
Removed this as it is confusing and prone to errors. New change will rely on partition count of reference version only